### ISSUE TITLE
Regression: CSS output differs from Less.js for mixin selector

### ISSUE THREAD

**Issue by fuf13** [wikimedia/less.php#108]
Version: 4.2.0+

### Description
A CSS class used as a mixin that has an extra definition with an attribute selector will cause all properties to be merged at the point of use of the mixin. See example below.
This behavior is unexpected and is different from the JS version.
In version 4.1.1, the behaviour was correct.

### Source LESS code
``` less
.btn {
   color: blue;
   font-weight: bold;

   &[title] {
      background: lightgreen;
   }
}
.btn[disabled] {
   cursor: not-allowed;
   color: red;
}
.btn[test] {
   background-color: pink;
}

.btn-specific {
   .btn();
}
```

### Current CSS output (version 4.2.0+)
``` css
.btn {
   color: blue;
   font-weight: bold;
} 
.btn[title] {
   background: lightgreen;
} 
.btn[disabled] {
   cursor: not-allowed;
   color: red;
} 
.btn[test] {
   background-color: pink;
} 
.btn-specific {
   color: blue;
   font-weight: bold;
   cursor: not-allowed;
   color: red;
   background-color: pink;
} 
.btn-specific[title] {
   background: lightgreen;
}
``` 

### Expected CSS output
``` css
.btn {
   color: blue;
   font-weight: bold;
} 
.btn[title] {
   background: lightgreen;
} 
.btn[disabled] {
   cursor: not-allowed;
   color: red;
} 
.btn[test] {
   background-color: pink;
} 
.btn-specific {
   color: blue;
   font-weight: bold;
}
.btn-specific[title] {
   background: lightgreen;
}
```

**Comment by Krinkle** [comment]
Thank you @fuf13 , this is a regression in 4.2.0. I've imported the issue at <https://phabricator.wikimedia.org/T363076> and we'll release a fix soon.

**Comment by Krinkle** [comment]
I believe this is now fixed by commit c047afebcf08e1f84e10a002fdbb16fac72f0006 in latest master. We'll cut a release and pubilsh to Packagist later this week.

Feel free to verify this by installing `"wikimedia/less.php": "dev-master#1f5fcc8224de83658780560112bd8ba7b541d949"` in composer.json. Or, test your input file via the `bin/lessc` command in a clone of this repository.

**Comment by Krinkle** [comment]
Released in v4.3.0. https://github.com/wikimedia/less.php/blob/v4.3.0/CHANGES.md
